src/nupic/engine/Link.cpp

```diff
@@ -396,17 +407,164 @@ void Link::write(LinkProto::Builder& proto) const
   proto.setSrcOutput(srcOutputName_.c_str());
   proto.setDestRegion(destRegionName_.c_str());
   proto.setDestInput(destInputName_.c_str());
+
+  // Save dealayed outputs
+  auto delayedOutputsBuilder = proto.initDelayedOutputs(propagationDelay_);
+  for (size_t i=0; i < propagationDelay_; ++i)
+  {
+    const Array & array = srcBuffer_[i];
+
+    auto genericArrayBuilder = delayedOutputsBuilder[i];
+
+    const size_t elementCount = array.getCount();
+
+    const auto arrayType = array.getType();
+
+    switch (arrayType)
```

Comment:
@scottpurdy: No, because `copyArrayToArrayProto` is a templated function that relies on the source buffer type and builder type to be pre-determined before the call to 'copyArrayToArrayProto`.